apps/system/test/unit/task_manager_test.js

```diff
@@ -1,1377 +1,570 @@
+      tm.hide().then(() => tm.stop()).then(done, done);
+      clock.tick(1000);
     });
-    suite('after destroying all the cards', function() {
-      setup(function() {
-        sendAppTerminated(apps.sms);
-        sendAppTerminated(apps.game);
-      });
 
-      test('should go back home', function(done) {
-        var stub = this.sinon.stub(home, 'open');
+    test('Proper state', (done) => {
+      assert.isTrue(tm.isShown());
+      assert.ok(document.querySelector('#cards-view.active:not(.empty)'));
 
-        waitForEvent(window, 'cardviewclosed').then(function() {
-          assert.isTrue(stub.calledOnce, 'home was open');
-        }, failOnReject)
-        .then(function() { done(); }, done);
+      assert.equal(
+        MockStackManager.getCurrent(),
+        tm.currentCard.app);
 
-        sendAppTerminated(apps.game);
-        fakeFinish(this.sinon.clock, home);
-      });
+      done();
     });
-  });
 
-  suite('app is killed', function() {
-    setup(function() {
-      MockStackManager.mStack = [
-        appConfigs.sms,
-        appConfigs.game
-      ].map((config) => {
-        var app = new AppWindow(config);
-        apps[app.name] = app;
-        return app;
+    test('Wheel handling for accessibility', (done) => {
+      var card = tm.currentCard;
+      var killAppStub = sinon.stub(card.app, 'kill');
+      sinon.stub(card.app, 'killable', () => true);
+      tm.handleEvent({
+        type: 'wheel',
+        deltaMode: 2,
+        DOM_DELTA_PAGE: 2,
+        deltaY: 1
       });
-      MockStackManager.mCurrent = 0;
-      showTaskManager(this.sinon.clock);
-    });
+      assert.isTrue(killAppStub.called);
 
-    test('removeCard is called on appterminated', function() {
-      var deadApp = apps.game;
-      var card = taskManager.cardsByAppID[deadApp.instanceID];
-      var removeCardSpy = this.sinon.spy(taskManager, 'removeCard');
-      var destroySpy = this.sinon.spy(card, 'destroy');
-      sendAppTerminated(deadApp);
-      this.sinon.clock.tick();
-
-      assert.isTrue(removeCardSpy.calledOnce, 'removeCard was called');
-      assert.isTrue(destroySpy.calledOnce, 'card.destroy was called');
-      assert.equal(cardsList.childNodes.length, 1);
+      done();
     });
-  });
 
-  suite('exit >', function() {
-    setup(function() {
-      MockStackManager.mStack = [
-        appConfigs.sms,
-        appConfigs.game
-      ].map((config) => {
-        var app = new AppWindow(config);
-        apps[app.name] = app;
-        return app;
-      });
-      MockStackManager.mCurrent = 0;
-      this.sinon.stub(AppWindow.prototype, 'open', function() {
-        this.element.dispatchEvent(new CustomEvent('_opened'));
+    test('wheel left/right event', function() {
+      this.sinon.spy(tm, 'updateScrollPosition');
+      tm.handleEvent({
+        type: 'wheel',
+        deltaMode: 2,
+        DOM_DELTA_PAGE: 2,
+        deltaX: 1
       });
+      assert.ok(tm.updateScrollPosition.calledOnce);
```

Comment:
instead of simply spying I think we should inspect the aria attributes on the dom here